src/actors/FilterBarActor.js

```diff
@@ -0,0 +1,88 @@
+import * as SharedUtils from '../utils/SharedUtils';
+
+export class FilterBarActor {
+  constructor(filterBarStore, tableStore) {
+    this.filterBarStore = filterBarStore;
+    this.tableStore = tableStore;
+  }
+
+  getSavedSearches() {
+    return this.filterBarStore.getSavedSearches() || [];
+  }
+
+  getFilter(filterUid) {
+    return this.filterBarStore.getFilter(filterUid)
+  }
+
+  enableFilter(filterUid, value) {
+    this.filterBarStore.enableFilter(filterUid, value);
+  }
+
+  disableFilter(filterUid) {
+    this.filterBarStore.disableFilter(filterUid);
+  }
+
+  disableAllFilters() {
+    this.filterBarStore.disableAllFilters();
+    this.applyFilters();
+  }
+
+  updateFilter(filterUid, propKey, propValue) {
+    this.filterBarStore.updateFilter(filterUid, propKey, propValue)
+  }
+
+  getEnabled() {
+    return this.filterBarStore.getEnabled();
+  }
+
+  getDisabled() {
+    return this.filterBarStore.getDisabled();
+  }
+
+  applyFilters() {
+    var id = this.filterBarStore.getId();
+    var searchUrl = this.filterBarStore.getSearchUrl();
+    var queryObject = this.filterBarStore.getQuery();
+
+    if (this.filterBarStore.persistent) {
+      SharedUtils.updateUrl('q', queryObject);
+    }
+
+    this.tableStore.setUrl(window.location.href);
```

Comment:
This the actual fetch could be done as an action. Fetching in the store could lead to race conditions as you mentioned earlier.